### PR TITLE
Make edgeHub metrics go through asp.net

### PR DESCRIPTION
**Pull request by lfitchett**
This fixes edgeHub metrics on windows needing to be run as an admin.

### CODE REVIEW

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Hosting.cs [hunk 1]
```diff
@@ -51,6 +51,8 @@ public class Hosting
                                         SslProtocols = sslProtocols
                                     });
                             });
+
+                        options.Listen(!Socket.OSSupportsIPv6 ? IPAddress.Any : IPAddress.IPv6Any, 9600);
```
Comment (kkdawkins) [inline edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Hosting.cs]: Is there ever a reason why we would want to change this port? I see that above, they are pulling a port from a configuration (with 443 as the default). Should you follow the same pattern?

File: test/modules/TwinTester/TwinEdgeOperationsInitializer.cs [hunk 2]
```diff
@@ -36,7 +36,6 @@ public static async Task<TwinEdgeOperationsInitializer> CreateAsync(RegistryMana
 
                 // reset reported properties
                 await TwinTesterUtil.ResetTwinReportedPropertiesAsync(moduleClient, twin);
-
```
Comment (kkdawkins) [inline test/modules/TwinTester/TwinEdgeOperationsInitializer.cs]: Nit: Revert this file's change? It just pollutes the git history.

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Hosting.cs (hunk 1 above)
Comment (varunpuranik) [reply edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Hosting.cs]: +1

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Program.cs [hunk 3]
```diff
@@ -68,9 +69,8 @@ static async Task<int> MainAsync(IConfigurationRoot configuration)
             logger.LogInformation($"OptimizeForPerformance={configuration.GetValue("OptimizeForPerformance", true)}");
             logger.LogInformation("Loaded server certificate with expiration date of {0}", certificates.ServerCertificate.NotAfter.ToString("o"));
 
-            var metricsListener = container.Resolve<IMetricsListener>();
             var metricsProvider = container.Resolve<IMetricsProvider>();
-            Metrics.Init(metricsProvider, metricsListener, logger);
+            Metrics.Init(metricsProvider, new NullMetricsListener(), logger);
```
Comment (varunpuranik) [inline edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Program.cs]: Remove

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Microsoft.Azure.Devices.Edge.Hub.Service.csproj [hunk 4]
```diff
@@ -42,6 +42,8 @@
     <PackageReference Include="Microsoft.Extensions.Configuration" Version="2.2.0" />
     <PackageReference Include="Microsoft.Extensions.Configuration.EnvironmentVariables" Version="2.2.0" />
 	<PackageReference Include="Autofac.Extensions.DependencyInjection" Version="4.3.1" />
+	<PackageReference Include="prometheus-net" Version="3.4.0" />
```
Comment (varunpuranik) [inline edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Microsoft.Azure.Devices.Edge.Hub.Service.csproj]: I would like to not take a dependency on the prometheus-net lib from Service

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Startup.cs [hunk 5]
```diff
@@ -44,6 +45,12 @@ public IServiceProvider ConfigureServices(IServiceCollection services)
 
         public void Configure(IApplicationBuilder app)
         {
+            // Seperate metrics endpoint from https authentication
+            app.Map("/metrics", metricsApp =>
+            {
+                metricsApp.UseMetricServer(string.Empty);
```
Comment (varunpuranik) [inline edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Startup.cs]: Is this a prometheus-net extension? Can we hide behind one of our types?

File: test/modules/TwinTester/TwinEdgeOperationsInitializer.cs (hunk 2 above)
Comment (varunpuranik) [reply test/modules/TwinTester/TwinEdgeOperationsInitializer.cs]: +1

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Microsoft.Azure.Devices.Edge.Hub.Service.csproj (hunk 4 above)
Comment (lfitchett) [reply edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Microsoft.Azure.Devices.Edge.Hub.Service.csproj]: spoke in person: 
prometheus-net.AspNetCore has a dependency on both prometheus-net and aspnet core. Either edgeHub service must take a dependency on Prometheus, or Util must take a dependency on asp.net. 

We decided service should take the dependency.

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Startup.cs (hunk 5 above)
Comment (lfitchett) [reply edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/Startup.cs]: same as above

File: test/modules/TwinTester/TwinEdgeOperationsInitializer.cs [hunk 6]
```diff
@@ -36,8 +36,6 @@ public static async Task<TwinEdgeOperationsInitializer> CreateAsync(RegistryMana
 
                 // reset reported properties
                 await TwinTesterUtil.ResetTwinReportedPropertiesAsync(moduleClient, twin);
-                
-
                 return new TwinEdgeOperationsInitializer(registryManager, moduleClient, reporter, 0);
```
Comment (kkdawkins) [inline test/modules/TwinTester/TwinEdgeOperationsInitializer.cs]: Please revert the change to this file...

Comment (kkdawkins) [reply test/modules/TwinTester/TwinEdgeOperationsInitializer.cs]: I believe that the best way to do this is to squash your change, do a `git reset HEAD~1` and then checkout this file.
